src/context/watchLater-context.js

```diff
@@ -0,0 +1,57 @@
+import { createContext, useContext, useState } from "react";
+import axios from "axios";
+
+const WatchLaterContext = createContext();
+
+const WatchLaterProvider = ({ children }) => {
+  const [watchLaterVideos, setWatchLaterVideos] = useState([]);
+
+  const handleAddToWatchLater = async (video) => {
+    try {
+      const token = localStorage.getItem("token");
+      const response = await axios.post(
+        "/api/user/watchlater",
+        { video },
+        {
+          headers: {
+            authorization: token,
+          },
+        }
+      );
+      setWatchLaterVideos(response.data.watchlater);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const handleRemoveFromWatchLater = async (id) => {
+    try {
+      const token = localStorage.getItem("token");
```

Comment:
suggestion - you can extract out this line into a separate function for reuse purposes.

Reply:
Yaah alert and toast I will add later and make changes for above suggesstions, thanks

Reply:
done